enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,494 @@
+to kube-apiserver (and other API servers) many version ago. There are two configuration mechanisms for the 
+audit policy:
+
+1. [**dynamic audit**](https://github.com/kubernetes/enhancements/blob/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/sig-auth/0014-dynamic-audit-configuration.md): an alpha API which lets the 
+user to create [`AuditSink`](https://github.com/kubernetes/api/blob/1fc28ea2498c5c1bc60693fab7a6741b0b4973bc/auditregistration/v1alpha1/types.go#L65) objects in the cluster with webhook backends which will receive the events.
+2. [**static file-based policies**](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy): policies in the master host file system passed via flags to the API servers.
+
+The former is an alpha API supposed to replace the latter file-based policies. I.e. the latter is supposed
+to go away eventually.
+
+The former policies are very restricted in its current form by only allowing to set 
+- the [**audit level**](https://github.com/kubernetes/api/blob/1fc28ea2498c5c1bc60693fab7a6741b0b4973bc/auditregistration/v1alpha1/types.go#L102) of events (how much is logged, e.g. meta data only or complete requests or responses)
+- the [**stages**](https://github.com/kubernetes/api/blob/1fc28ea2498c5c1bc60693fab7a6741b0b4973bc/auditregistration/v1alpha1/types.go#L106) audit events are created in (request received, response started, response finished).
+
+The [latter policies](https://github.com/kubernetes/apiserver/blob/822585c65b3829cc20fbcbe76481f7413d5fd98e/pkg/apis/audit/v1/types.go#L153) are
+- full-featured
+- with a pretty [advanced rule system](https://github.com/kubernetes/apiserver/blob/822585c65b3829cc20fbcbe76481f7413d5fd98e/pkg/apis/audit/v1/types.go#L163) with complex semantics.
+
+In OpenShift 3.x the file-based policies were customizable by customers. In 4.x, with its unclear future and its complexity involved, 
+we hesitate to go the same route: if the file-based API goes away, it is infeasible to patch it back into our OpenShift API server
+binaries without major effort.
+
+From the requirements we get from customers, it is apparent that the file-based policies are far too low-level, and
+too complex to get right. This enhancement is about this high-level API which we can offer to satisfy the needs of
+customers with strong regulatory or security requirement, while being easy to configure for them and feasible to 
+maintain in the future.
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### Goals
+
+- add an abstract audit policy configuration to `apiservers.config.openshift.io/v1`
+  - with few predefined policies
+  - easily extensible in the future 
+  - few knobs on the level of concrete OpenShift customer requirement
+- applied by kube-apiserver, openshift-apiserver and oauth-apiserver.
+
+### Non-Goals
```

Comment:
non-goal: minimal sets of metadata.  Customers can prune before long term storage if they wish.